code/modules/spells/spell_types/undirected/bloodrage.dm

```diff
@@ -0,0 +1,32 @@
+/datum/action/cooldown/spell/undirected/bloodrage
+	name = "Bloodrage"
+	desc = "Grants you unbound strength for a short while."
+	button_icon_state = "bloodrage"
+	sound = 'sound/magic/bloodrage.ogg'
+
+	has_visual_effects = FALSE
+	spell_type = SPELL_MIRACLE
+	antimagic_flags = MAGIC_RESISTANCE_HOLY
+	associated_skill = /datum/skill/magic/holy
+	invocation = "GRAGGAR!! GRAGGAR!! GRAGGAR!!"
+	invocation_type = INVOCATION_SHOUT
+	charge_required = FALSE
+	cooldown_time = 5 MINUTES
+	spell_cost = 80
+	var/static/list/purged_effects = list(
+			/datum/status_effect/incapacitating/immobilized,
+			/datum/status_effect/incapacitating/paralyzed,
+			/datum/status_effect/incapacitating/stun,
+			/datum/status_effect/incapacitating/knockdown,
+		)
+
+/datum/action/cooldown/spell/undirected/bloodrage/cast(mob/living/cast_on)
```

Comment:
add before_cast check for if they are dead, and this should unbuckle before everything in cast